platform/ext/target/nordic_nrf/nrf5340/target_cfg.c

```diff
@@ -0,0 +1,191 @@
+
+#ifdef BL2
+    .secondary_partition_base =
+        (uint32_t)&REGION_NAME(Load$$LR$$, LR_SECONDARY_PARTITION, $$Base),
+
+    .secondary_partition_limit =
+        (uint32_t)&REGION_NAME(Load$$LR$$, LR_SECONDARY_PARTITION, $$Base) +
+        SECONDARY_PARTITION_SIZE - 1,
+#endif /* BL2 */
+};
+
+/* To write into AIRCR register, 0x5FA value must be write to the VECTKEY field,
+ * otherwise the processor ignores the write.
+ */
+#define SCB_AIRCR_WRITE_MASK ((0x5FAUL << SCB_AIRCR_VECTKEY_Pos))
+
+enum tfm_plat_err_t enable_fault_handlers(void)
+{
+    /* Explicitly set secure fault priority to the highest */
+    NVIC_SetPriority(SecureFault_IRQn, 0);
+
+    /* Enables BUS, MEM, USG and Secure faults */
+    SCB->SHCSR |= SCB_SHCSR_USGFAULTENA_Msk
+                  | SCB_SHCSR_BUSFAULTENA_Msk
+                  | SCB_SHCSR_MEMFAULTENA_Msk
+                  | SCB_SHCSR_SECUREFAULTENA_Msk;
+    return TFM_PLAT_ERR_SUCCESS;
+}
+
+enum tfm_plat_err_t system_reset_cfg(void)
+{
+    uint32_t reg_value = SCB->AIRCR;
+	
+    /* Clear SCB_AIRCR_VECTKEY value */
+    reg_value &= ~(uint32_t)(SCB_AIRCR_VECTKEY_Msk);
+
+    /* Enable system reset request only to the secure world */
+    reg_value |= (uint32_t)(SCB_AIRCR_WRITE_MASK | SCB_AIRCR_SYSRESETREQS_Msk);
+
+    SCB->AIRCR = reg_value;
+
+    return TFM_PLAT_ERR_SUCCESS;
+}
+
+enum tfm_plat_err_t init_debug(void)
+{
+#if defined(DAUTH_NONE)
+    /* Disable debugging */
+    NRF_CTRLAP->APPROTECT.DISABLE = 0;
+    NRF_CTRLAP->SECUREAPPROTECT.DISABLE = 0;
+#elif defined(DAUTH_NS_ONLY)
+    /* Allow debugging Non-Secure only */
+    NRF_CTRLAP->APPROTECT.DISABLE = CTRLAPPERI_APPROTECT_DISABLE_KEY_Msk;
+    NRF_CTRLAP->SECUREAPPROTECT.DISABLE = 0;
+#elif defined(DAUTH_FULL)
+    /* Allow debugging */
+    NRF_CTRLAP->APPROTECT.DISABLE = CTRLAPPERI_APPROTECT_DISABLE_KEY_Msk;
+    NRF_CTRLAP->SECUREAPPROTECT.DISABLE = CTRLAPPERI_SECUREAPPROTECT_DISABLE_KEY_Msk;
+#elif !defined(DAUTH_CHIP_DEFAULT)
+#error "No debug authentication setting is provided."
+#endif
+    /* Lock access to APPROTECT, SECUREAPPROTECT */
+    NRF_CTRLAP->APPROTECT.LOCK = CTRLAPPERI_APPROTECT_LOCK_LOCK_Locked <<
+        CTRLAPPERI_APPROTECT_LOCK_LOCK_Msk;
+    NRF_CTRLAP->SECUREAPPROTECT.LOCK = CTRLAPPERI_SECUREAPPROTECT_LOCK_LOCK_Locked <<
+        CTRLAPPERI_SECUREAPPROTECT_LOCK_LOCK_Msk;
+
+    return TFM_PLAT_ERR_SUCCESS;
+}
+
+/*----------------- NVIC interrupt target state to NS configuration ----------*/
+enum tfm_plat_err_t nvic_interrupt_target_state_cfg(void)
+{
+    /* Target every interrupt to NS; unimplemented interrupts will be Write-Ignored */
+    for (uint8_t i = 0; i < sizeof(NVIC->ITNS) / sizeof(NVIC->ITNS[0]); i++) {
+        NVIC->ITNS[i] = 0xFFFFFFFF;
+    }
+
+    /* Make sure that the SPU is targeted to S state */
+    NVIC_ClearTargetState(SPU_IRQn);
+
+    return TFM_PLAT_ERR_SUCCESS;
+}
+
+/*----------------- NVIC interrupt enabling for S peripherals ----------------*/
```

Comment:
Which 5 peripherals?

Reply:
That's an "S" I think this sentence was copy-pasted from other targets

Reply:
Haha, right 😅